File: 0001_0100/0042/0042.c
#include <stdio.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n;
	long long k = 1;

	fscanf(f, "%d", &n);
	while (n > 4)
		k *= 3, n -= 3;
	fprintf(q, "%I64d", k * n);
	return 0;
}

File: 0001_0100/0042/0042.go
package main

import (
	"fmt"
	"os"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var n, k int64
	fmt.Fscan(f, &n)
	for k = 1; n > 4; n -= 3 {
		k *= 3
	}
	fmt.Fprint(q, n*k)
}
